sdks/python/apache_beam/io/iobase.py

```diff
@@ -1424,9 +1454,9 @@ def process(
           restriction_tracker=core.DoFn.RestrictionParam(
               _SDFBoundedSourceWrapper._SDFBoundedSourceRestrictionProvider(
                   source, chunk_size))):
-        start_pos, end_pos = restriction_tracker.current_restriction()
-        range_tracker = self.source.get_range_tracker(start_pos, end_pos)
-        return self.source.read(range_tracker)
+        for output in restriction_tracker.get_tracking_source().read(
```

Comment:
Nit: Any reason not to simply return `restriction_tracker.get_tracking_source().read(...)`?

Reply:
`return` directly will result in reading a whole block at once, which makes `read` less granular.

Reply:
Actually it'd return the underlying iterator, which would be read in just as granular a fraction as this loop.

Reply:
yes you are right. Changed back in commit https://github.com/apache/beam/pull/8944/commits/917846f9cc910198c4c6a5b0da9c05b6d3f0f196